packages/dict/src/dict.ts

```diff
@@ -0,0 +1,54 @@
+import { Predicate, Guard, Reducer, UnaryFn } from '@proem/function'
+
+export interface Dictionary {
+  [key:string]: any
+}
+
+export const map = <A extends Dictionary, B>(dict: A, mapfn: ([k, v]: [string, any]) => B): B[] => {
```

Comment:
Since both `map` and `filter` loop over the entries to produce a new result, they could be defined in terms of `reduce`. In my opinion, it would result in a bit nicer looking code.